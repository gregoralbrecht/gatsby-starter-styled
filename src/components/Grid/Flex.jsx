import styled, { flexbox, propTypes } from '@style'

import Box from './Box'

const Flex = styled(Box)`
  ${flexbox}
`

Flex.displayName = `Flex`

Flex.defaultProps = {
  display: `flex`,
}

Flex.propTypes = {
  ...propTypes.flexbox,
}

export default Flex
