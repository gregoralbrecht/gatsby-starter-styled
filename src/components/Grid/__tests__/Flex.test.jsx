import React from 'react'
import { render } from '@test'
import Flex from '../Flex'
import 'jest-styled-components'

test('<Flex> renders', () => {
  const { getByText } = render(<Flex>Hello world</Flex>)

  expect(getByText(`Hello world`)).toMatchSnapshot()
  expect(getByText(`Hello world`)).toHaveStyle(`display: flex`)
})

test('<Flex> renders with props', () => {
  const { getByText } = render(
    <Flex m={2} px={[1, 2, 3]}>
      Hello world
    </Flex>
  )

  expect(getByText(`Hello world`)).toMatchSnapshot()
  expect(getByText(`Hello world`)).toHaveStyle(`margin: 8px`)
})

test('<Flex> renders with as prop', () => {
  const { getByText } = render(
    <Flex m={2} as="header">
      Hello world
    </Flex>
  )

  expect(getByText(`Hello world`)).toMatchSnapshot()
  expect(getByText(`Hello world`)).toHaveStyle(`margin: 8px`)
})
