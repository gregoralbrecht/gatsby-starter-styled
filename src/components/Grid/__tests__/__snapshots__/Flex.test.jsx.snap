// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Flex> renders 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div
  class="c0"
  display="flex"
>
  Hello world
</div>
`;

exports[`<Flex> renders with as prop 1`] = `
.c0 {
  margin: 8px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<header
  class="c0"
  display="flex"
>
  Hello world
</header>
`;

exports[`<Flex> renders with props 1`] = `
.c0 {
  margin: 8px;
  padding-left: 4px;
  padding-right: 4px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

@media screen and (min-width:576px) {
  .c0 {
    padding-left: 8px;
    padding-right: 8px;
  }
}

@media screen and (min-width:768px) {
  .c0 {
    padding-left: 16px;
    padding-right: 16px;
  }
}

<div
  class="c0"
  display="flex"
>
  Hello world
</div>
`;
